R/setops.R

```diff
@@ -103,8 +103,8 @@ fsetequal = function(x, y, all=TRUE) {
 # all.equal ----
 
 all.equal.data.table = function(target, current, trim.levels=TRUE, check.attributes=TRUE, ignore.col.order=FALSE, ignore.row.order=FALSE, tolerance=sqrt(.Machine$double.eps), ...) {
-  stopifnot(is.logical(trim.levels), is.logical(check.attributes), is.logical(ignore.col.order), is.logical(ignore.row.order), is.numeric(tolerance))
-  if (!is.data.table(target) || !is.data.table(current)) stop("'target' and 'current' must both be data.tables")
+  stopifnot(is.logical(trim.levels), is.logical(check.attributes), is.logical(ignore.col.order), is.logical(ignore.row.order), is.numeric(tolerance), is.data.table(target))
+  if (!is.data.table(current)) return(paste0('target is data.table, current is ', data.class(current)))
```

Comment:
what about `check.attributes=FALSE`?

Reply:
`all.equal(data.frame(a=1), c(a=1), check.attributes = F)` returns TRUE -- do you think it should for `data.table` as well? 

The problem with continuing on despite `current` not being a data.table (could be as simple as a numeric, in that example) is that you have to a lot more careful about calling `setcolorder`, `nrow`, etc, on it which happens in the next few lines. Hence my original idea to just call `NextMethod`. I saw this as an easy middle ground. Maybe if `check.attributes = FALSE` just do `current <- as.data.table(current)` then keep going??

Reply:
class is an attribute. `NextMethod` could eventually be an option but for a `check.attributes=FALSE` then. Also we need to take care and check if any extra arguments haven't been used, because they will be ignored in a non-data.table all.equal method.

Reply:
Ok, I'll try to think about this some more. As is, I do believe it is still an improvement, though clearly not perfect.

Reply:
It is improvement, but I don't think we should merge it like this without support for `check.attributes` argument.